src/coreclr/src/jit/codegenarm.cpp

```diff
@@ -998,13 +998,38 @@ void CodeGen::genCodeForStoreLclFld(GenTreeLclFld* tree)
     // Ensure that lclVar nodes are typed correctly.
     assert(!varDsc->lvNormalizeOnStore() || targetType == genActualType(varDsc->TypeGet()));
 
-    GenTree*    data = tree->gtOp1;
-    instruction ins  = ins_Store(targetType);
-    emitAttr    attr = emitTypeSize(targetType);
+    GenTree* data = tree->gtOp1;
+    emitAttr attr = emitTypeSize(targetType);
 
     assert(!data->isContained());
     genConsumeReg(data);
-    emit->emitIns_S_R(ins, attr, data->GetRegNum(), varNum, offset);
+    regNumber dataReg = data->GetRegNum();
+    if (varTypeIsFloating(targetType) && ((offset % emitTypeSize(TYP_FLOAT)) != 0))
+    {
+        // Arm supports unaligned access only for integer types,
+        // convert the storing floating data into 2 integer registers and write them as int.
+        regNumber addr = tree->ExtractTempReg();
+        emit->emitIns_R_S(INS_lea, attr, addr, varNum, offset);
+        if (targetType == TYP_FLOAT)
+        {
+            regNumber floatAsInt = tree->GetSingleTempReg();
+            emit->emitIns_R_R(INS_vmov_f2i, EA_4BYTE, floatAsInt, dataReg);
+            emit->emitIns_R_R(INS_str, EA_4BYTE, floatAsInt, addr);
+        }
+        else
+        {
+            regNumber halfdoubleAsInt1 = tree->ExtractTempReg();
```

Comment:
If you don't mind going one step further this could be optimized downto
```suggestion
            regNumber halfdoubleAsInt1 = tree->ExtractTempReg();
            regNumber halfdoubleAsInt2 = tree->GetSingleTempReg();
            assert (halfdoubleAsInt1 < halfdoubleAsInt2);
            emit->emitIns_R_R_R(INS_vmov_d2i, EA_8BYTE, halfdoubleAsInt1, halfdoubleAsInt2, dataReg);
            regMaskTP doubleAsIntMask = genRegMask(halfdoubleAsInt1) | genRegMask(halfdoubleAsInt2);
            emit->emitIns_R_I(INS_stm, EA_4BYTE, addr, doubleAsIntMask);
```

Reply:
Not worth doing this now